src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -0,0 +1,340 @@
+                // we cannot choose one entity as the referencing entity. This is because for a referencing entity,
+                // the values for all the referencing fields should be derived from the insertion in the referenced entity.
+                // However, here both entities contain atleast one relationship field whose value is provided in the request.
+                if (doesSourceBodyContainAnyRelationshipField && doesTargetBodyContainAnyRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"The relationship fields can be specified either for the source entity: {sourceEntityName} or the target entity: {targetEntityName} " +
+                        $"for the relationship: {relationshipName} at level: {nestingLevel}.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // The source/target entities can assume a value for insertion for a relationship field if:
+                // 1. The field is autogenerated, or
+                // 2. The field is given a non-null value by the user - since we don't allow null values for a relationship field.
+                bool canSourceAssumeThisFieldValue = isSourceRelationshipColumnAutogenerated || sourceColumnvalue is not null;
+                bool canTargetAssumeThisFieldValue = isTargetRelationshipColumnAutogenerated || targetColumnvalue is not null;
+
+                // Update whether all the values(non-null) for relationship fields are available for source/target.
+                canSourceAssumeAllRelationshipFieldValues = canSourceAssumeAllRelationshipFieldValues && canSourceAssumeThisFieldValue;
+                canTargetAssumeAllRelationshipFieldsValues = canTargetAssumeAllRelationshipFieldsValues && canTargetAssumeThisFieldValue;
+
+                // If the values for all relationship fields cannot be assumed for neither source nor target,
+                // the multiple create request cannot be executed.
+                if (!canSourceAssumeAllRelationshipFieldValues && !canTargetAssumeAllRelationshipFieldsValues)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Neither source entity: {sourceEntityName} nor the target entity: {targetEntityName} for the relationship: {relationshipName} at level: {nestingLevel} " +
+                        $"provide sufficient data to perform a multiple-create (related insertion) on the entities.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+            }
+
+            return canSourceAssumeAllRelationshipFieldValues ? targetEntityName : sourceEntityName;
```

Comment:
> via insertion to the referencED entity 

No, this concept of assuming a value via insertion in the referenced entity comes during request validation when we have actually determined which entity acts as the referenced entity. Here, assuming a value means:
1. Either the column is autogenerated, or
2. The column is assigned a value by the user

> via insertion to the referencED entity or the value is user provided, right?

For determining the referencing/referenced entity, we rely on the user giving the values for relationship fields. If one of the entity has the values for all the relationship fields, that entity will be considered as the referenced entity. If source can assume value for all relationship fields, it will act as the referenced entity and the target will act as the referencing entity and vice-versa.

Reply:
resolved over call. clarified intent, comments are reflective of what we discussed.